nearswap/src/deposit.rs

```diff
@@ -191,6 +191,122 @@ impl AccountDeposit {
 //   we don't do the storage refunds, instead we shold accumulate what storage has been used and keeping the following invariant all the time: account_deposit.amount >= account_deposit.storage  * STORAGE_PRICE_PER_BYTE
 // +
 
-// TODO make unit tests for AccountDeposit
 #[cfg(test)]
-mod tests {}
+mod tests {
+    use super::AccountDeposit;
+    
+    #[test]
+    fn add_works() {
+        let mut deposit = AccountDeposit {
+            near: 12,
+            storage_used: 10,
+            tokens:[("token1".to_string(), 100),
+            ("token2".to_string(), 50)]
+            .iter().cloned().collect()
+        };
+
+        AccountDeposit::add(&mut deposit, &"token1".to_string(), 10);
+        assert_eq!(deposit.tokens.get(&"token1".to_string()), Some(&110));
+    }
+
+    #[test]
+    fn add_new_works() {
+        let mut deposit = AccountDeposit {
+            near: 12,
+            storage_used: 10,
+            tokens:[("token1".to_string(), 100)]
+            .iter().cloned().collect()
+        };
```

Comment:
let create a helper function to create `AccountDeposit`. this line is almost repeated:
```
fn new_account_deposit(extra_tokens: Vec<String, u64>) -> AccountDeposit {
   AccountDepoist { near: 12, storage_used: 10, tokens: [("token1".to_string(), 100)].concatenate(....)...
}
```

Reply:
ok